OMCompiler/toJulia/MMToJuliaTestSuite/syntaxCheck.jl

```diff
@@ -0,0 +1,109 @@
+#=
+# This file is part of OpenModelica.
+#
+# Copyright (c) 1998-2019, Open Source Modelica Consortium (OSMC),
+# c/o Linköpings universitet, Department of Computer and Information Science,
+# SE-58183 Linköping, Sweden.
+#
+# All rights reserved.
+#
+# THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
+# THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+# ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+# RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+# ACCORDING TO RECIPIENTS CHOICE.
+#
+# The OpenModelica software and the Open Source Modelica
+# Consortium (OSMC) Public License (OSMC-PL) are obtained
+# from OSMC, either from the above address,
+# from the URLs: http://www.ida.liu.se/projects/OpenModelica or
+# http://www.openmodelica.org, and in the OpenModelica distribution.
+# GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+#
+# This program is distributed WITHOUT ANY WARRANTY; without
+# even the implied warranty of  MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+# IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+#
+# See the full OSMC Public License conditions for more details.
+#
+=#
+
+module SyntaxTest
+#=
+#    This part of the tests checks that the files generated by MMTojulia.dumpProgram complies
+#    to the syntactic rules of Julia. No semantic check nor runtime check.
+=#
+using Test
+include("../metaModelicaToJulia.jl")
+
+#= Macro that returns true if a function does not throw =#
+macro false_if_throw(expr)
+  quote
+    try
+      $expr
+      true
+    catch e
+      false
+    end
+  end |> esc
+end
+
+macro test_nothrow_nowarn(expr)
+  quote
+    @test_nowarn $expr
+  end |> esc
+end
+
+function executeTestSteps(homeDirectory, sourceDirectory, outputDirectory, omc)
+  outDir = createDirectoryReportErrorOnFailure(abspath(outputDirectory))
+  translateFilesIfOutputIsEmpty(sourceDirectory, outDir, omc, homeDirectory)
+  checkSyntax(outDir, sourceDirectory)
+  cd(homeDirectory)
+end
+
+function syntaxCheck(omc)
```

Comment:
This test currently does two things. It both tests the syntax and if AbsynToJulia.tpl translates without throwing errors. Currently, my algorithms test is failing. I plan to add a test for translating Graphviz at some point when ./Algorithm is translated correctly. 

When the syntax tests passes I will add semantic tests that tests these different functions